CI: Use custom container for E2E tests

Myestery opened the pull request:
## Summary
Use a pre-built container image with all dependencies for Playwright E2E tests, eliminating ~130s setup time per shard.

## Changes
- Use `ghcr.io/comfy-org/comfyui-ci-container:0.0.8` container for test jobs
- Container includes: Playwright browsers, Node.js, pnpm, Python, ComfyUI backend (v0.5.1), all Python deps
- Simplified setup: just copy devtools and start server (no cloning, no pip install, no browser setup)
- Add `version-bump*` to `branches-ignore` to skip E2E tests for version bump PRs
- Replace cache with artifacts (cache doesn't work inside containers)

## Benefits
- ~130s faster per shard (no ComfyUI clone, no pip install, no browser download)
- Consistent environment across all test jobs
- Simpler workflow configuration

## Container Image
Repository: https://github.com/comfy-org/comfyui-ci-container
Image: `ghcr.io/comfy-org/comfyui-ci-container:0.0.8`

## Test plan
- [x] Verify CI workflow runs with container
- [x] Verify Playwright tests pass
- [x] Verify snapshot updates work correctly

christian-byrne reviewed (dismissed):
**For future PRs:**

Unused files needing removal:

- .github/actions/setup-playwright/action.yml

Docs needing updates after this PR:

- ~~browser_tests/README.md - Remove pnpm exec playwright install chromium --with-deps~~ keep this as it's docs for running locally
- CONTRIBUTING.md - Update Playwright test section to mention Docker approach

Myestery committed:
CI: Use custom container for E2E tests

- Use ghcr.io/myestery/comfyui-test-action:latest container
- Container includes: Playwright, Node.js, pnpm, Python, ComfyUI backend
- Simplified setup: just copy devtools and start server directly
- No more cloning ComfyUI or pip install in each job
- Add version-bump* to branches-ignore

Myestery committed:
use action from comfy Org

Myestery commented on .github/workflows/ci-tests-e2e.yaml:
```diff
@@ -28,23 +26,21 @@ jobs:
     needs: setup
     runs-on: ubuntu-latest
+    container:
+      image: ghcr.io/myestery/comfyui-test-action:latest
```
I now use pwuser

Myestery committed:
Merge main into ci/playwright-docker-containers

christian-byrne commented on .github/workflows/ci-tests-e2e.yaml:
```diff
@@ -6,7 +6,7 @@ on:
   pull_request:
     branches-ignore:
-      [wip/*, draft/*, temp/*, vue-nodes-migration, sno-playwright-*]
+      [wip/*, draft/*, temp/*, vue-nodes-migration, sno-playwright-*, version-bump*]
```
I think we should still run on version bumps

christian-byrne commented on .github/workflows/ci-tests-e2e.yaml:
```diff
@@ -94,39 +88,37 @@ jobs:
     needs: setup
     runs-on: ubuntu-latest
+    container:
+      image: ghcr.io/myestery/comfyui-test-action:latest
```
Is this correct?

christian-byrne commented on .github/workflows/ci-tests-e2e.yaml:
```diff
@@ -94,39 +88,37 @@ jobs:
+        run: |
+          cp -r ./tools/devtools/* /ComfyUI/custom_nodes/ComfyUI_devtools/
+          cd /ComfyUI && python3 main.py --cpu --multi-user --front-end-root $GITHUB_WORKSPACE/dist &
+          wait-for-it --service 127.0.0.1:8188 -t 600
```
If this is being re-used 3 times, can we make it a separate action?

Myestery committed:
fix: update CI workflow to refine branch ignore patterns and correct container image

Myestery commented:
after this gets merged we need to update all snapshots once

Myestery commented on .github/workflows/ci-tests-e2e.yaml:
```diff
@@ -28,24 +26,22 @@ jobs:
     runs-on: ubuntu-latest
     timeout-minutes: 60
+    container:
+      image: ghcr.io/comfy-org/comfyui-ci-container:0.0.3
```
The container already uses the pw user. see https://github.com/Comfy-Org/comfyui-ci-container/blob/main/Dockerfile

Myestery committed:
update to use correct GCR image

Myestery committed:
use pnpm caching

DrJKL commented on .github/workflows/ci-tests-e2e.yaml:
```diff
@@ -54,27 +50,25 @@ jobs:
+      - name: Checkout repository
+        uses: actions/checkout@v5
+      - name: Download built frontend
+        uses: actions/download-artifact@v4
```
This is up to v7, why are we using v4?

Myestery committed:
fix: add packages:read permission for ghcr.io container pull

Add packages:read permission to jobs using the comfyui-ci-container
image to fix Docker pull denied errors in GitHub Actions.

Myestery committed:
fix: add container credentials for ghcr.io authentication

Use GITHUB_TOKEN to authenticate when pulling the private container
image from ghcr.io.

Myestery committed:
build: Update CI container image to version 0.0.4 in workflows

Myestery committed:
ci: update container image to 0.0.5 and remove ComfyUI server setup from e2e workflow.

Myestery committed:
refactor: extract ComfyUI server startup into reusable action

Create .github/actions/start-comfyui-server composite action to
deduplicate server startup logic across container-based workflows.

Myestery committed:
fix: correct misleading cache comment in workflow files

Myestery committed:
Merge main into ci/playwright-docker-containers

Myestery committed:
fix: use GITHUB_WORKSPACE env var and trim verbose output

Myestery committed:
fix: add front_end_root input and fail-fast shell behavior

Myestery committed:
chore: update CI container image to 0.0.6

Myestery committed:
chore: Update CI container image version to 0.0.7 in workflows.

Myestery committed:
chore: Update CI container image to version 0.0.8.

Commit:
[automated] Update test expectations
